device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -355,6 +355,26 @@ else if (subscriptionType == DEVICE_OPERATION_TWIN_SUBSCRIBE_DESIRED_PROPERTIES_
                         || senderLinkHandler instanceof AmqpsTwinSenderLinkHandler && messageType == DEVICE_TWIN
                         || senderLinkHandler instanceof AmqpsMethodsSenderLinkHandler && messageType == DEVICE_METHODS)
                 {
+                    if (messageType == DEVICE_TWIN)
+                    {
+                        if (explicitInProgressTwinSubscriptionMessage != null)
+                        {
+                            // Don't send any twin messages while a twin subscription is in progress. Wait until the subscription
+                            // has been acknowledged by the service before sending it.
```

Comment:
Q - from where will this message be retried?

Reply:
The function that calls this checks to see if the message send was successful. If it wasn't, it requeues the message. that's why we return ```false``` here to mimic a sending failure